Patch for https://github.com/cython/cython/pull/1687
Only needed for Cygwin systems

commit 632f18f2710eaf6ee2b3a3aaa0749c7886299e64
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Thu Apr 27 17:13:39 2017 +0200

    Do not use special dll linkage for "cdef public" functions

diff --git a/Cython/Compiler/ModuleNode.py b/Cython/Compiler/ModuleNode.py
index 6356e78..76fe4e5 100644
--- a/Cython/Compiler/ModuleNode.py
+++ b/Cython/Compiler/ModuleNode.py
@@ -214,8 +214,7 @@ class ModuleNode(Nodes.Node, Nodes.BlockNode):
     def generate_public_declaration(self, entry, h_code, i_code):
         h_code.putln("%s %s;" % (
             Naming.extern_c_macro,
-            entry.type.declaration_code(
-                entry.cname, dll_linkage="DL_IMPORT")))
+            entry.type.declaration_code(entry.cname)))
         if i_code:
             i_code.putln("cdef extern %s" % (
                 entry.type.declaration_code(entry.cname, pyrex=1)))
@@ -2849,7 +2848,7 @@ def generate_cfunction_declaration(entry, env, code, definition):
             dll_linkage = "DL_IMPORT"
         elif entry.visibility == 'public':
             storage_class = Naming.extern_c_macro
-            dll_linkage = "DL_EXPORT"
+            dll_linkage = None
         elif entry.visibility == 'private':
             storage_class = "static"
             dll_linkage = None
